server/svix-server/src/queue/rabbitmq.rs

```diff
@@ -160,3 +160,69 @@ async fn declare_bound_queue(
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use futures::StreamExt as _;
+    use lapin::options::{BasicConsumeOptions, BasicQosOptions};
+    use svix_ksuid::{KsuidLike as _, KsuidMs};
+
+    use crate::{cfg, queue::QueueTask};
+
+    #[tokio::test]
+    // run with `cargo test -- --ignored rabbitmq` only when rabbitmq is up and configured
+    #[ignore]
+    async fn test_messages_have_ids() {
```

Comment:
I think the prior art for this sort of thing is to return immediately (i.e. pass the test) for configs that don't support the thing.

iirc we see this in cache-related tests, specifically checking for the "none" cache.

https://github.com/svix/svix-webhooks/blob/910f29b1667c2fef973f22ef7c8b779413642b39/server/svix-server/tests/it/worker.rs#L138

Not a huge difference but it'll save us from having to run the test binary twice for this one case.

Reply:
I quite dislike that, it makes it hard / impossible to tell whether the test actually ran and passed, or it saw a configuration that didn't match its expectations. I think we should change those tests to match this one ;)

Regarding the extra run of the test binary, note that `--ignored` only runs ignored tests, `--include-ignored` would also (re)run others.